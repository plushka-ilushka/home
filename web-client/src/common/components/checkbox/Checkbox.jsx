import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './checkbox.scss';

export default function Checkbox(props) {
  const {
    title,
    className,
    name,
    type,
    onClick,
    ...setProps
  } = props;

  return (
    <label className={classNames('labels i', className)} htmlFor={name}>
      {title}
      <input
        className={classNames('check-box', className)}
        id={name}
        name={name}
        type={type}
        onClick={onClick}
        {...setProps}
      />
      <i />
    </label>
  );
}

Checkbox.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.bool,
};

Checkbox.defaultProps = {
  title: '',
  type: 'checkbox',
  className: undefined,
  id: '',
  name: '',
  onClick: undefined,
};
