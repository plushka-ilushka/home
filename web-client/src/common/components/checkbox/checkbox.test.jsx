import React from 'react';
import { mount } from 'enzyme';
import Checkbox from './Checkbox';

describe('Checkbox Component', () => {
  it('Checkbox Snapshot', () => {
    expect(mount(<Checkbox />)).toMatchSnapshot();
  });
});
