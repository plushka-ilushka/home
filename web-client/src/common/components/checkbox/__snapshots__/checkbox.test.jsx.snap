// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Checkbox Snapshot 1`] = `
<Checkbox
  id=""
  name=""
  title=""
  type="checkbox"
>
  <label
    className="labels i"
    htmlFor=""
  >
    <input
      className="check-box"
      id=""
      name=""
      type="checkbox"
    />
    <i />
  </label>
</Checkbox>
`;
